src/core.rs

```diff
@@ -574,14 +583,30 @@ impl<T> Webauthn<T> {
 
         // If user verification is required for this assertion, verify that the User Verified bit of
         // the flags in authData is set.
-        match policy {
-            UserVerificationPolicy::Required => {
+        //
+        // We also check the historical policy too. See designs/authentication-use-cases.md
+
+        match (&policy, &cred.registration_policy) {
+            (UserVerificationPolicy::Required, _) => {
+                // If we requested required, enforce that.
                 if !data.authenticator_data.user_verified {
                     return Err(WebauthnError::UserNotVerified);
                 }
             }
-            UserVerificationPolicy::Preferred_DO_NOT_USE | UserVerificationPolicy::Discouraged => {}
-        };
+            (UserVerificationPolicy::Discouraged, UserVerificationPolicy::Discouraged) => {
+                // If this token always verifies, even under discouraged, we can enforce that.
+                if cred.verified {
+                    // This token always sends UV, so we enforce that.
+                    if !data.authenticator_data.user_verified {
```

Comment:
I think this might cause some false positive errors. perhaps we want to control this with a config?
i.e: `if cred.verified && self.config.get_require_uv_consistency()`?

Reply:
Yep, this is a good point. I'll add that switch now.